### ISSUE TITLE
Per Diem Rates lookup - Can we use != vs a = for looking up a record

### ISSUE THREAD

**Issue by JDEKid** [GSA/GSA-APIs#63]

**Comment by JDEKid** [comment]
Hi,

We would like to lookup for Non-standard rate for a Zip. I know there are a couple fields that tell you whether the rate is Standard or Non-Standard - "City" : "Standard Rate" or "DestinationID" : 0.

Can I use a **_Not Equal To_** in my lookup? I know the other option is to get all and then parse through the fetched records. Just makes my code complex and dirty for what I need.

Thanks.

?resource_id=8ea44bc4-22ba-4386-b84c-1494ab28964b&limit=300000&filters=%7B"FiscalYear":"2019","City":"Standard%20Rate","Zip":"35006"%7D

**Comment by adborden** [comment]
Right, I don't think there is a way to use "not equal to" with the [datastore_search](https://docs.ckan.org/en/2.5/maintaining/datastore.html#ckanext.datastore.logic.action.datastore_search) API, however, the more complex and powerful [datastore_search_sql](https://docs.ckan.org/en/2.5/maintaining/datastore.html#ckanext.datastore.logic.action.datastore_search_sql) allows you to write the query you're after. Here's a `curl` command that get's what you're after, I believe. `<>` is synonymous with `!=`.

It just uses PostgreSQL behind the scenes,  so column names with capitals must be double quoted. Make sure to use a library to handle your URL encoding, since it can get pretty ugly if you have to manually encode everything by hand.

```
curl -v 'https://inventory.data.gov/api/action/datastore_search_sql' --get --data-urlencode sql="select * from \"8ea44bc4-22ba-4386-b84c-1494ab28964b\" where \"FiscalYear\" = 2019 AND \"Zip\" = 35006 AND \"City\" <> 'Standard Rate'"
```

Please try it out and let me know if it works for you.
